starlette/responses.py

```diff
@@ -452,8 +470,30 @@ async def _handle_multiple_ranges(
                     }
                 )
 
-    def _should_use_range(self, http_if_range: str) -> bool:
-        return http_if_range == self.headers["last-modified"] or http_if_range == self.headers["etag"]
+    def _is_not_modified(self, http_if_none_match: str | None, http_if_modified_since: str | None) -> bool:
+        """
+        Given the request and response headers, return `True` if an HTTP
+        "Not Modified" response could be returned instead.
+        """
+        if http_if_none_match is not None:
+            match = [tag.strip(" W/") for tag in http_if_none_match.split(",")]
```

Comment:
There are some issues here; the strong and weak semantics of etag should not be handled this way.